internal/cmd/config/cmd.go

```diff
@@ -12,11 +13,25 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
+type CorsConfig struct {
+	AllowedOrigins         []string                                  `yaml:"allowed-origins"`
+	AllowOriginFunc        func(origin string) bool                  `yaml:"allow-origin-func"`
+	AllowOriginRequestFunc func(r *http.Request, origin string) bool `yaml:"allow-origin-request-func"`
```

Comment:
2021 - Software engineer Namit invented new revolutionary way to deserialize  functions from json

Reply:
> April 2021 - Software engineer Namit invented new revolutionary way to deserialize functions from json

Lol, I saw this coming 😂

The thing is that this PR is still `Work in Progress` and there are a couple of more changes I wanted to make in this PR before requesting for a review.
Since it was bedtime for me back then, I left those changes for today (sorry, forgot to mention this in the PR description).

The first change that I had planned for today was to fix the function extraction part from yaml (which is what you have pointed out too) and the second was to change the default values of some of the fields, like currently the `AllowedOrigins` field takes a default value of `[ ]` instead of `["*"]`, which needs to be changed.

Will push a commit that fixes both of these in some time.